src/main/java/ch/cern/eam/wshub/core/tools/InforFieldTools.java

```diff
@@ -129,6 +130,16 @@ public InforFieldTools(CustomFieldsTools customFieldsTools, Tools tools) {
             } else if ("UserDefinedFields".equals(wshubField.getAnnotation(InforField.class).xpath()[0]) ||
                        "StandardUserDefinedFields".equals(wshubField.getAnnotation(InforField.class).xpath()[0])) {
                 wshubField.set(wshubObject, transformInforObject(new UserDefinedFields(), inforValue));
+            } else if(List.class.isAssignableFrom(inforValue.getClass())){
+                List inforValueAsList = (List) inforValue;
+                if(inforValueAsList.size() == 0) return;
```

Comment:
If an empty array is received, should it not initialize an empty list?

Reply:
Fixed. Also handled edge case when List received is null.